From fd029511d9262df0c49560a6e479429329d6a1e6 Mon Sep 17 00:00:00 2001
From: "qiankun.wang" <qiankun.wang@amlogic.com>
Date: Fri, 12 May 2023 15:01:39 +0800
Subject: [PATCH] cpu: fix core0 cannot be offline in 32BIT kernel [1/1]

PD#SWPL-114632

Problem:
fix core0 cannot be offline in 32BIT kernel

Solution:
fix core0 cannot be offline in 32BIT kernel

Verify:
on s4_aq222

Change-Id: I0c1bf91e013f133851494622bd59da33ac038a51
Signed-off-by: qiankun.wang <qiankun.wang@amlogic.com>
---
 arch/arm/kernel/smp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/kernel/smp.c b/arch/arm/kernel/smp.c
index 0cdaebb28b26..083662ead0a8 100644
--- a/arch/arm/kernel/smp.c
+++ b/arch/arm/kernel/smp.c
@@ -236,7 +236,11 @@ int platform_can_hotplug_cpu(unsigned int cpu)
 	 * since this is special on a lot of platforms, e.g. because
 	 * of clock tick interrupts.
 	 */
+#if IS_ENABLED(CONFIG_AMLOGIC_HOTPLUG_ARM_CPU0)
+	return 1;
+#else
 	return cpu != 0;
+#endif
 }
 
 static void ipi_teardown(int cpu)
-- 
2.25.1

